Ical.Net/Evaluation/RecurringEvaluator.cs

```diff
@@ -85,24 +82,22 @@ protected HashSet<Period> EvaluateRDate(IDateTime referenceDate, DateTime period
     /// <param name="referenceDate"></param>
     /// <param name="periodStart">The beginning date of the range to evaluate.</param>
     /// <param name="periodEnd">The end date of the range to evaluate.</param>
-    protected HashSet<Period> EvaluateExRule(IDateTime referenceDate, DateTime periodStart, DateTime periodEnd)
+    protected IEnumerable<Period> EvaluateExRule(IDateTime referenceDate, DateTime? periodStart, DateTime? periodEnd)
     {
         if (Recurrable.ExceptionRules == null || !Recurrable.ExceptionRules.Any())
-        {
-            return new HashSet<Period>();
-        }
+            return [];
 
-        var exRuleEvaluatorQuery = Recurrable.ExceptionRules.SelectMany(exRule =>
+        var exRuleEvaluatorQueries = Recurrable.ExceptionRules.Select(exRule =>
         {
             var exRuleEvaluator = exRule.GetService(typeof(IEvaluator)) as IEvaluator;
             if (exRuleEvaluator == null)
             {
                 return Enumerable.Empty<Period>();
             }
             return exRuleEvaluator.Evaluate(referenceDate, periodStart, periodEnd, false);
-        });
+        }).ToList();
```

Comment:
same as above
(no deferred execution in order to convert the result to a list to ensure that all recurrence rules are fully evaluated before any further processing?)